:book: Add detailed instructions for Weather Agent demo (#78)
kagenti/kagenti: docs/demo-weather-agent.md

# Weather Agent Demo

> 

Reviewer: This doc needs a preamble pointing to the current `docs/cn-demos.md` doc to say "this is a prerequisite." 

Also can we rename cn-demos.md? The vision is all the demos start from that place no?

This document provides detailed steps for running the **Weather Agent** demo proof-of-concept (PoC).

In the [Import New Agent](#import-new-agent) section, you'll build and deploy the [`acp_weather_service`](https://github.com/kagenti/agent-examples/tree/main/acp/acp_weather_service) agent.

In the [Import New Tool](#import-new-tool) section, you'll build and deploy the [`weather_tool`](https://github.com/kagenti/agent-examples/tree/main/mcp/weather_tool) tool.

> **Prerequisites:**  
> Ensure you've completed the Kagenti platform setup as described in the [Installation](../cn-demos.md#installation) section. This demo uses `ACP` protocol, so you will not need `OPENAI_API_KEY` in your environment setup.

> You should also open the Agent Platform Demo Dashboard as instructed in the [Run the Demo](../cn-demos.md#run-the-demo) section.

Reviewer 2: I think this is basic enough that you can just put the link here. It seems unnecessary to force users to go to another page just for this.

---

## Import New Agent

To deploy the Weather Agent:

1. Navigate to [Import New Agent](http://kagenti-ui.localtest.me:8080/Import_New_Agent#import-new-agent) in the Kagenti UI.
2. In the **Select Namespace to Deploy Agent** drop-down, choose the `<namespace>` where you'd like to deploy the agent. (These namespaces are defined in your `.env` file.)
3. Under [**Select Environment Variable Sets**](http://kagenti-ui.localtest.me:8080/Import_New_Agent#select-environment-variable-sets), select:
   - `mcp-weather`

>    - `ollama`

Reviewer: ```suggestion
   - ONE OF `ollama` or `openai`. This determines whether the agent will talk to an ollama endpoint or the OpenAI API. Note that the OpenAI option requires that the `OPEN_API_KEY` be populated with your OpenAI API token
```

Author: I am not sure about this suggestion. I have tried various steps end never got the OpenAI to work. I supposed there are more steps required, and we would have to include them in order to properly handle OpenAI. We should try to compare our environments first

Author: Can we please make this a separate PR that provides a detailed instructions on running OpenAI? 
I just opened an issue #84 to address it

4. In the **Agent Source Repository URL** field, use the default:
   <https://github.com/kagenti/agent-examples>
   Or use a custom repository accessible using the GitHub ID specified in your `.env` file.
5. For **Git Branch or Tag**, use the default `main` branch (or select another as needed).
6. Set **Protocol** to `acp`.
7. Under [**Specify Source Subfolder**](http://kagenti-ui.localtest.me:8080/Import_New_Agent#specify-source-subfolder):
   - Click `Select from examples`
   - Choose: `acp/acp_weather_service`
8. Click **Build New Agent** to deploy.

---

## Import New Tool

To deploy the Weather Tool:

1. Navigate to [Import New Tool](http://kagenti-ui.localtest.me:8080/Import_New_Tool#import-new-tool) in the UI.
2. Select the same `<namespace>` as used for the agent.
3. In the **Select Environment Variable Sets** section, select:
   - `mcp-weather`
   - `ollama`
4. Use the same source repository:
   <https://github.com/kagenti/agent-examples>
5. Choose the `main` branch or your preferred branch.
6. Set **Select Protocol** to `MCP`.
7. Under **Specify Source Subfolder**:
   - Select: `mcp/weather_tool`
8. Click **Build New Tool** to deploy.

---

## Validate The Deployment

> Depending on your hosting environment, it may take some time for the deployment to complete.

Reviewer 2: ```suggestion
Depending on your hosting environment, it may take some time for the agent and tool deployments to complete.
```

   ```

3. Tail the logs to ensure both services have started successfully.
   For the agent:

   ```console
    installer$ kubectl -n team1 logs -f acp-weather-service-8bb4644fc-4d65d
    Defaulted container "acp-weather-service" out of: acp-weather-service, kagenti-client-registration (init)
    INFO:     Started server process [18]
    INFO:     Waiting for application startup.
    INFO:     Application startup complete.
    INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
    ```

    For the tool:
    ```console
    installer$ kubectl -n team1 logs -f weather-tool-5bb675dd7c-ccmlp
    Defaulted container "weather-tool" out of: weather-tool, kagenti-client-registration (init)
    INFO:     Started server process [19]
    INFO:     Waiting for application startup.
    INFO:     Application startup complete.
    INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
    ```

4. Once you see the logs indicating that both services are up and running, you're ready to proceed to [Authorize the Agent and the Tool](#authorize-the-agent-and-the-tool).

## Authorize the Agent and the Tool

> Once the agent and the tool are deployed, you need to authorize the agent to interact with the tool:

Reviewer: What we do here right now is not technically "Authorizing" the agent (recall the tool does no validation right now)
Instead, I might adjust these texts to say we are "registering" the agent onto Keycloak.

Reviewer: we don't need to worry too much because I suppose we will be updating this demo soon anyhow

Author: This is a foundation for further updates. We will revisit Authorization section again later.

1. Navigate to the **Admin** section in the Kagenti UI.
2. Under [**Identity Management (Keycloak)**](http://kagenti-ui.localtest.me:8080/Admin#identity-management-keycloak), click the **Go to Identity Management Console** link.  
   This will open the Keycloak service in a new window or tab.

> 3. Log in with the default admin demo credentials:
>    Username: admin
>    Password: admin

Reviewer 2: ```suggestion
3. Log in with the default admin demo credentials:
   Username: `admin`
   Password: `admin`
```

> 4. In the top-left hamburger menu,ensure the **Keycloak** realm is selected as the *Current realm*.  

Reviewer 2: ```suggestion
4. In the top-left hamburger menu, ensure the **Keycloak** realm is selected as the *Current realm*.  
```

   If not, click **Manage realms** and select **Keycloak Master**.
5. Under the **Manage** section in the left menu, click **Clients**.
6. Locate the `weather-agent` client in the list and click on it.
7. On the client settings page, ensure the **Enabled** toggle is switched on.  
This authorizes the Weather Agent to access the Weather Tool.

> 8. Return to the [Run the Weather Agent Demo](#run-the-weather-agent-demo) section to test the setup.

Reviewer 2: ```suggestion
8. Return to the [Run the Weather Agent Demo](#run-the-weather-agent-demo) section to test the setup.
```

9. You can experiment with disabling this client to observe how authorization impacts the agent’s ability to function.

---

> ## Run the Weather Agent Demo

Reviewer: ```suggestion
## Chat with the Enabled Weather Agent
```

Once the deployment is complete and the Agent is authorized to access the Tool, you can run the demo:

1. Navigate to the **Agent Catalog** in the Kagenti UI.
2. Select the same `<namespace>` used during the agent deployment.
3. Under [**Available Agents in <namespace>**](http://kagenti-ui.localtest.me:8080/Agent_Catalog#available-agents-in-kagenti-system), select `acp-weather-service` and click **View Details**.
4. Scroll to the bottom of the page. In the input field labeled *Say something to the agent...*, enter:

   ```console
   What is the weather in NY?
   ```

> 5. You will see *Agent Thinking...* message. Depending on the speed of your hosting environment, the agent will return a weather response. For example:

Reviewer 2: ```suggestion
5. You will see the *Agent Thinking...* message. Depending on the speed of your hosting environment, the agent will return a weather response. For example:
```

   ```console
     The current weather in NY is mostly sunny with a temperature of 22.6 degrees Celsius (73.07 degrees Fahrenheit). There is a gentle breeze blowing at 6.8 km/h (4.2 mph) from the northwest. It's currently daytime, and the weather code indicates fair weather with no precipitation.
    ```

6. You can tail the log files (as shown in the [Validate the Deployment section](#validate-the-deployment)) to observe the interaction between the agent and the tool in real time.

## Cleanup

Currently, the Kagenti UI does not provide built-in cleanup capabilities for agents or tools.
However, you can manually remove them by deleting their Custom Resources (CRs) from the cluster.

### Step 1: List Custom Resource Definitions (CRDs)

```console
    installer$ kubectl get crds | grep kagenti
    components.kagenti.operator.dev             2025-07-23T23:11:59Z
    platforms.kagenti.operator.dev              2025-07-23T23:11:59Z
```

### Step 2: List deployed components in your namespace

```console
    installer$ kubectl get components.kagenti.operator.dev -n team1
    NAME                  SUSPEND
    acp-weather-service   false
    weather-tool          false
```

> ### Step3: Delete the Agent and the Tool

Reviewer: ```suggestion
### Step 3: Delete the Agent and the Tool
```
